aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -114,9 +140,99 @@ macro_rules! unwrap_or_discard {
+                        storage_gas_schedule.per_byte_read.into();
+                    gas_params.natives.table.common.load_failure = 0.into();
+                },
+                7.. => {
+                    gas_params.natives.table.common.load_base_legacy = 0.into();
+                    gas_params.natives.table.common.load_base_new =
+                        storage_gas_schedule.per_item_read.into();
+                    gas_params.natives.table.common.load_per_byte =
+                        storage_gas_schedule.per_byte_read.into();
+                    gas_params.natives.table.common.load_failure = 0.into();
+                },
+                _ => (),
+            }
+        }
+
+        let storage_gas_params = StorageGasParameters::new(
+            gas_feature_version,
+            gas_params.as_ref(),
+            storage_gas_schedule.as_ref(),
+        );
+
+        // TODO(Gas): Right now, we have to use some dummy values for gas parameters if they are not found on-chain.
+        //            This only happens in a edge case that is probably related to write set transactions or genesis,
+        //            which logically speaking, shouldn't be handled by the VM at all.
+        //            We should clean up the logic here once we get that refactored.
+        let (native_gas_params, abs_val_size_gas_params) = match &gas_params {
+            Some(gas_params) => (gas_params.natives.clone(), gas_params.misc.abs_val.clone()),
+            None => (
+                NativeGasParameters::zeros(),
+                AbstractValueSizeGasParameters::zeros(),
+            ),
+        };
+
+        let features = Features::fetch_config(&storage).unwrap_or_default();
+
+        // If no chain ID is in storage, we assume we are in a testing environment and use ChainId::TESTING
+        let chain_id = ChainId::fetch_config(&storage).unwrap_or_else(ChainId::test);
+
+        let timestamp = ConfigurationResource::fetch_config(&storage)
+            .map(|config| config.last_reconfiguration_time())
+            .unwrap_or(0);
+
+        let mut timed_features = TimedFeatures::new(chain_id, timestamp);
+        if let Some(profile) = crate::AptosVM::get_timed_feature_override() {
+            timed_features = timed_features.with_override_profile(profile)
+        }
+
+        let inner = MoveVmExt::new(
+            native_gas_params,
+            abs_val_size_gas_params,
+            gas_feature_version,
+            chain_id.id(),
+            features.clone(),
+            timed_features,
+        )
+        .expect("should be able to create Move VM; check if there are duplicated natives");
+
+        let mut vm = Self {
+            move_vm: inner,
+            gas_feature_version,
+            gas_params,
+            storage_gas_params,
+            version: None,
+            transaction_validation: None,
+            features,
+        };
+        vm.version = Version::fetch_config(&storage);
```

Comment:
why are we assigning these members here right after creation? wouldn't Self { ... vesion: Version::fetch_config, transaction_validation: Self::get_transaction_validation(..)..} work?

or we could assign to version and transaction_validation variables earlier (like inner)